src/app/workouts/workoutCreate/WorkoutCreate.tsx

```diff
@@ -0,0 +1,142 @@
+// src/pages/WorkoutCreateFromRoutine.tsx
+
+import React, { useEffect, useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { Workout } from '@/types/entities/Workout';
+import { RoutineExercise } from '@/types/entities/Routine';
+import { createWorkout } from '@/api/workoutApi';
+import { AppRoutes } from '@/types/routes';
+import {
+  Button,
+  Typography,
+  List,
+  ListItem,
+  ListItemText,
+  Divider, Grid,
+} from '@mui/material';
+
+//TODO: to be removed
+import { mockRoutineData } from '@/types/entities/mockRoutineData';
+
+const WorkoutCreate: React.FC = () => {
+  const location = useLocation();
+  const navigate = useNavigate();
+  const [workout, setWorkout] = useState<Workout | null>(null);
```

Comment:
you are sending routine: data via navigate function, why dont you set that routine data as initial state?

Reply:
good point! 
will do

Reply:
I kinda do in useEffect(), the data just has different format, so I basically map it